src/Roslyn.Diagnostics.Analyzers/Core/AbstractDoNotCopyValue.cs

```diff
@@ -178,6 +238,140 @@ public void Dispose()
+            }
+
+            public override void VisitAlias(IAliasSymbol symbol)
+            {
+                base.VisitAlias(symbol);
+            }
+
+            public override void VisitArrayType(IArrayTypeSymbol symbol)
+            {
+                base.VisitArrayType(symbol);
+            }
+
+            public override void VisitAssembly(IAssemblySymbol symbol)
+            {
+                base.VisitAssembly(symbol);
+            }
+
+            public override void VisitDiscard(IDiscardSymbol symbol)
+            {
+                base.VisitDiscard(symbol);
+            }
+
+            public override void VisitDynamicType(IDynamicTypeSymbol symbol)
+            {
+                base.VisitDynamicType(symbol);
+            }
+
+            public override void VisitEvent(IEventSymbol symbol)
+            {
+                base.VisitEvent(symbol);
+            }
+
+            public override void VisitField(IFieldSymbol symbol)
+            {
+                // Fields of copyable value types must be copyable
+                if (Cache.IsNonCopyableType(symbol.Type)
+                    && !Cache.IsNonCopyableType(symbol.ContainingType)
+                    && symbol.ContainingType.IsValueType)
+                {
+                    _context.ReportDiagnostic(symbol.CreateDiagnostic(NoFieldOfCopyableTypeRule, symbol.Type, symbol));
+                }
+
+                base.VisitField(symbol);
+            }
+
+            public override void VisitFunctionPointerType(IFunctionPointerTypeSymbol symbol)
+            {
+                base.VisitFunctionPointerType(symbol);
+            }
+
+            public override void VisitLabel(ILabelSymbol symbol)
+            {
+                base.VisitLabel(symbol);
+            }
+
+            public override void VisitLocal(ILocalSymbol symbol)
+            {
+                base.VisitLocal(symbol);
+            }
+
+            public override void VisitMethod(IMethodSymbol symbol)
+            {
+                base.VisitMethod(symbol);
+            }
+
+            public override void VisitModule(IModuleSymbol symbol)
+            {
+                base.VisitModule(symbol);
+            }
+
+            public override void VisitNamedType(INamedTypeSymbol symbol)
+            {
+                base.VisitNamedType(symbol);
+            }
+
+            public override void VisitNamespace(INamespaceSymbol symbol)
+            {
+                base.VisitNamespace(symbol);
+            }
+
+            public override void VisitParameter(IParameterSymbol symbol)
+            {
+                base.VisitParameter(symbol);
+            }
+
+            public override void VisitPointerType(IPointerTypeSymbol symbol)
+            {
+                base.VisitPointerType(symbol);
+            }
+
+            public override void VisitProperty(IPropertySymbol symbol)
+            {
+                // Auto-properties cannot have non-copyable types
```

Comment:
probably worth commenting why?

Reply:
The user-facing description of the rule includes the explanation. The getter for an auto-property always copies the backing field, which is not allowed for a non-copyable type.

Reply:
I updated the code to include more complete descriptions. The user facing messages still have descriptions as well.